src/modules/apps/e2e/index.tsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react'
+import { render } from 'ink'
+
+import { apps, tester } from '../../../clients'
+import { RealTimeReport } from './report'
+import { getToken } from '../../../conf'
+import { ManifestEditor } from '../../../lib/manifest/ManifestEditor'
+import { ErrorBoundary } from './errorBoundary'
+import { TestRequest } from '../../../clients/Tester'
+
+
+export default async options => {
+  new EndToEndCommand(options).run()
+}
+
+class EndToEndCommand {
+  options : any
+
+  constructor(options) {
+    this.options = options
+  }
+  public run() {
+    if(this.options.workspace) {
+      return this.runWorkspaceTests()
+    }
+
+    return this.runAppTests()
+  }
+
+  private async runAppTests() {
+    const manifestEditor = await ManifestEditor.getManifestEditor()
+    const cleanAppId = manifestEditor.appLocator
+
+    const appList = await apps.listApps()
+    const appId = appList.data.map(({ app }) => app).find(app => app.startsWith(cleanAppId))
+  
+    if (appId === undefined) {
+      throw new Error('App was not found!')
+    }
+  
+    const testRequest = this.options.report
+      ? null
+      : await tester.test(
+          { integration: true, monitoring: true, authToken: this.options.token ? getToken() : undefined },
+          appId
+        )
+  
+    this.render(testRequest)
+  }
+
+  private async runWorkspaceTests() {
+    const testRequest = this.options.report
+      ? null
+      : await tester.test(
+          { integration: true, monitoring: true, authToken: this.options.token ? getToken() : undefined }
+        )
+    
+    this.render(testRequest)
+  }
+
+  private async render( testRequest: TestRequest ) {
+    const testId = testRequest ? testRequest.testId : (this.options.report as string)
```

Comment:
`testRequest` is null in case `this.options.report` is not defined, right? So what happens to `testId` if `report` is not given as a flag? It will always be `null | undefined`?

Reply:
Maybe we could improve the help on this flag? https://github.com/vtex/toolbelt/blob/7d4aa994086765d751ab80661b8a4c6589c51f5a/src/modules/tree.ts#L294
Something like (you can improve it):
`Check the results and state of a previously started test given its ID`

I don't know if report is a good name for this flag, but I don't have any suggestions, so if you don't have any ideas on how we could improve this it's ok.